### PR TITLE
[Lease.Azure] Async task and options code cleanup

### PR DESCRIPTION
**Pull request by Arkatufus**
Possible fix for #1251

Still unsure that this is the actual cause of the failure, problem was not reproduced.

### CODE REVIEW

File: src/coordination/azure/Akka.Coordination.Azure/AzureLease.cs [hunk 1]
```diff
@@ -85,23 +85,19 @@ public AzureLease(ExtendedActorSystem system, AtomicBoolean leaseTaken, LeaseSet
...
-                    .ContinueWith(t =>
-                    {
-                        return t.Result switch
-                        {
-                            LeaseActor.LeaseReleased _ => true,
-                            LeaseActor.InvalidRequest req => throw new LeaseException(req.Reason),
-                            _ => false
-                        };
-                    });
+                var result = await _leaseActor.Ask(LeaseActor.Release.Instance, _timeout);
```
Comment (Arkatufus) [inline src/coordination/azure/Akka.Coordination.Azure/AzureLease.cs:94]: Convert code to async-await pattern

File: src/coordination/azure/Akka.Coordination.Azure/AzureLease.cs [hunk 2]
```diff
@@ -113,24 +109,20 @@ public override Task<bool> Release()
...
-                    {
-                        return t.Result switch
-                        {
-                            LeaseActor.LeaseAcquired _ => true,
-                            LeaseActor.LeaseTaken _ => false,
-                            LeaseActor.InvalidRequest req => throw new LeaseException(req.Reason),
-                            _ => false
-                        };
-                    });
+                var result = await _leaseActor.Ask(new LeaseActor.Acquire(leaseLostCallback), _timeout);
```
Comment (Arkatufus) [inline src/coordination/azure/Akka.Coordination.Azure/AzureLease.cs:118]: Convert code to async-await pattern

File: src/coordination/azure/Akka.Coordination.Azure/AzureLeaseOption.cs [hunk 3]
```diff
@@ -51,20 +49,20 @@ public override void Apply(AkkaConfigurationBuilder builder, Setup? s = null)
                 sb.AppendLine($"lease-operation-timeout = {LeaseOperationTimeout.ToHocon()}");
             sb.AppendLine("}");
 
+            builder.AddHocon(sb.ToString(), HoconAddMode.Prepend);
+            
+            if(AzureCredential is null && ServiceEndpoint is null)
+                return;
```
Comment (Arkatufus) [inline src/coordination/azure/Akka.Coordination.Azure/AzureLeaseOption.cs:55]: Code cleanup, shortcut `Apply` to not inject a `Setup` class if there is no need for it

File: src/coordination/azure/Akka.Coordination.Azure/AzureLeaseOption.cs [hunk 4]
```diff
@@ -51,20 +49,20 @@ public override void Apply(AkkaConfigurationBuilder builder, Setup? s = null)
...
-            builder.AddHocon(sb.ToString(), HoconAddMode.Prepend);
-
             var setup = new AzureLeaseSetup
             {
                 AzureCredential = AzureCredential,
                 ServiceEndpoint = ServiceEndpoint,
                 BlobClientOptions = BlobClientOptions,
-                ApiServiceRequestTimeout = ApiServiceRequestTimeout,
-                ConnectionString = ConnectionString,
-                ContainerName = ContainerName
```
Comment (Arkatufus) [inline src/coordination/azure/Akka.Coordination.Azure/AzureLeaseOption.cs:67]: Code cleanup, remove non-needed information from `Setup` class

File: src/coordination/azure/Akka.Coordination.Azure/Internal/AzureApiImpl.cs [hunk 5]
```diff
@@ -167,6 +167,7 @@ public async Task<LeaseResource> ReadOrCreateLeaseResource(string name)
             {
                 switch ((HttpStatusCode)e.Status)
                 {
+                    case HttpStatusCode.PreconditionFailed:
```
Comment (Arkatufus) [inline src/coordination/azure/Akka.Coordination.Azure/Internal/AzureApiImpl.cs:170]: Bug fix, we did not use any pre-condition in our code but the driver implicitly added a pre-condition internally, need to capture this exception.

File: src/coordination/azure/Akka.Coordination.Azure/AzureLease.cs (hunk 1 above)
Comment (Aaronontheweb) [reply src/coordination/azure/Akka.Coordination.Azure/AzureLease.cs:94]: LGTM

File: src/coordination/azure/Akka.Coordination.Azure/AzureLease.cs (hunk 2 above)
Comment (Aaronontheweb) [reply src/coordination/azure/Akka.Coordination.Azure/AzureLease.cs:118]: lgtm

File: src/coordination/azure/Akka.Coordination.Azure/Internal/AzureApiImpl.cs (hunk 5 above)
Comment (Aaronontheweb) [reply src/coordination/azure/Akka.Coordination.Azure/Internal/AzureApiImpl.cs:170]: Nice catch

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**